pkg/webhooks/sgx/sgx.go

```diff
@@ -40,9 +40,41 @@ const (
 	encl                = namespace + "/enclave"
 	epc                 = namespace + "/epc"
 	provision           = namespace + "/provision"
-	provisionAnnotation = namespace + "/needs-provision"
+	quoteProvAnnotation = namespace + "/quote-provider"
 )
 
+// getAesmdVolume determines whether aesmd is a sidecar or a DaemonSet
+// based on how many containers within a pod added aesmd-volume mount
+// and returns the pod volume source accordingly.
+func getAesmdVolume(aesmdMounts int32) *corev1.Volume {
+	switch aesmdMounts {
+	case 0:
+		return nil
+	case 1:
+		// the deployment is aesmd itself
```

Comment:
Hmm. So if a container has `sgx.intel.com/quote-provider: aesmd`, it's always assumed to be aesmd, if there's only one such container in a pod? Is this right?

Reply:
this check is currently faulty, let me fix it first

Reply:
the logic is fixed now. does it make better sense?